syncer/syncer.go

```diff
@@ -299,91 +303,63 @@ func (s *Syncer) synchronize(ctx context.Context) bool {
 		logger.Info("sync is already running, giving up")
 		return false
 	}
+	defer s.setSyncerIdle()
 
 	// no need to worry about race condition for s.run. only one instance of synchronize can run at a time
 	s.run++
 	logger.With().Info(fmt.Sprintf("starting sync run #%v", s.run),
-		log.String("sync_state", s.getSyncState().String()),
-		log.FieldNamed("current", s.ticker.GetCurrentLayer()),
-		log.FieldNamed("latest", s.mesh.LatestLayer()),
-		log.FieldNamed("processed", s.mesh.ProcessedLayer()))
+		log.Stringer("sync_state", s.getSyncState()),
+		log.Stringer("last_synced", s.getLastSyncedLayer()),
+		log.Stringer("current", s.ticker.GetCurrentLayer()),
+		log.Stringer("latest", s.mesh.LatestLayer()),
+		log.Stringer("processed", s.mesh.ProcessedLayer()))
 
 	s.setStateBeforeSync(ctx)
-	var (
-		vQueue chan types.LayerID
-		vDone  chan struct{}
-		// number of attempts to start a validation goroutine
-		attempt = 0
-		// whether the data sync succeed. validation failure is not checked by design.
-		success = true
-		// the last layer that is synced in this run
-		lastSynced = s.mesh.ProcessedLayer()
-	)
-
-	attemptFunc := func() bool {
+	syncFunc := func() bool {
```

Comment:
why do we not define it as a static function?

Reply:
this function is not strictly static. it needs struct member (mesh/ticker) and method (s.getLastSyncedLayer()/s.setLastSyncedLayer()). i don't think the code would be cleaner changing this to a static func.